src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const newSeparator = String(separator);
+
+    let joinString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let checkedElement = this[i];
+
+      if (checkedElement === null || checkedElement === undefined) {
+        checkedElement = '';
+      }
+
+      if (i === 0) {
+        joinString += checkedElement;
+      } else {
```

Comment:
you can add this element at the beginning and start iterating from index 1, no need an extra check